equipment-requests/src/Code.js

```diff
@@ -0,0 +1,198 @@
+  form.addListItem().setTitle('Desktop').setChoiceValues(availableDesktops);
+  form.addListItem().setTitle('Monitor').setChoiceValues(availableMonitors);
+
+  var sheets = ss.getSheets().forEach(function(sheet) {
+    if (sheet.getFormUrl() == ss.getFormUrl()) {
+      sheet.hideSheet();
+    }
+  });
+  // Start workflow on each form submit
+  ScriptApp.newTrigger('onFormSubmit_')
+      .forForm(form)
+      .onFormSubmit()
+      .create();
+  // Archive completed items every 5m.
+  ScriptApp.newTrigger('processCompletedItems_')
+      .timeBased()
+      .everyMinutes(5)
+      .create();
+}
+
+/**
+ * Cleans up the project (stop triggers, form submission, etc.)
+ */
+function cleanup_() {
+  var formUrl = SpreadsheetApp.getActiveSpreadsheet().getFormUrl();
+  if (!formUrl) {
+    return;
+  }
+  ScriptApp.getProjectTriggers().forEach(function(trigger) {
+    ScriptApp.deleteTrigger(trigger);
+  });
+  var form = FormApp.openByUrl(formUrl)
+      .deleteAllResponses()
+      .setAcceptingResponses(false);
+}
+
+/**
+ * Handle new form submissions to trigger the workflow.
+ *
+ * @param {Object} event - Form submit event
+ */
+function onFormSubmit_(event) {
+  var response = mapResponse_(event.response);
+  sendNewEquipmentRequestEmail_(response);
+  var equipmentDetails = Utilities.formatString('%s\n%s\n%s',
+      response['Laptop'],
+      response['Desktop'],
+      response['Monitor']);
+  var row = ['New',
+    '',
+    response['Desk location'],
+    response['Employee name'],
+    response['Desk location'],
+    equipmentDetails,
+    response['email']];
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
+  var sheet = ss.getSheetByName('Pending requests');
+  sheet.appendRow(row);
+}
+
+/**
+ * Sweeps completed events, notifying the requestors and archiving them
+ * to the completed
+ *
+ * @param {Object} event
+ */
+function processCompletedItems_() {
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
+  var pending = ss.getSheetByName('Pending requests');
+  var completed = ss.getSheetByName('Completed requests');
+  var rows = pending.getDataRange().getValues();
+  rows.forEach(function(row, index) {
+    var status = row[0];
+    if (status === 'Completed' || status == 'Cancelled') {
+      pending.deleteRow(index + 1);
+      completed.appendRow(row);
+      sendEquipmentRequestCompletedEmail_({
+        'Employee name': row[3],
+        'Desk location': row[4],
+        'email': row[6],
+      });
+    }
+  });
+}
+
+/**
+ * Sends email notifying team a new equipment request has been submitted.
+ *
+ * @param {Object} request - Request details
+ */
+function sendNewEquipmentRequestEmail_(request) {
+  var template = HtmlService.createTemplateFromFile('new-equipment-request.html');
+  template.request = request;
+  template.sheetUrl = SpreadsheetApp.getActiveSpreadsheet().getUrl();
+  var msg = template.evaluate();
+  GmailApp.sendEmail(REQUEST_NOTIFICATION_EMAIL, 'New equipment request', msg.getContent(), {
```

Comment:
Should we use MailApp instead since it's more secure?

Reply:
Is it?

Reply:
done